### ISSUE TITLE
Build plan / Deployment plan metadata capture

### ISSUE THREAD

**Issue by mriedel-pgdx** [jfrog/bamboo-artifactory-plugin#42]
I'm using v2.1.0 of the plugin with Bamboo 5.15.7 build 51521.

I noticed that when I use the Generic Deploy task from a build plan, it captures the appropriate metadata information (environment variables, build numbers, etc.) with the artifact when deploying to Artifactory.

However, when using a deployment plan that uses the artifact from the build plan, that build metadata information is not captured and supplied to Artifactory using the same Generic Deploy task.

Is it possible to have deployment plans also get the artifact's build metadata for inclusion in Artifactory?

**Comment by eyalbe4** [comment]
You're right @mriedel-pgdx. Deployment plans currently deploy the artifacts only. Build-info publication from a deployment is not currently supported.

**Comment by dcasnowdon** [comment]
I'd love to see this functionality also. I noticed also that the build information available via the bamboo plugin is somewhat sparse compared to what it looks like is available in Jenkins. Lastly, it seems that I can not use File Specs to define the source patterns and target path, etc, in a Deployment Plan. 

Is this functionality planned? Is there a better way to do this (via the jfrog CLI?)?

![screen shot 2018-07-06 at 7 43 33 pm](https://user-images.githubusercontent.com/3860429/42372187-ebe3dd4c-8154-11e8-945c-6fbf49352590.png)
